protoc-gen-connect-kotlin/src/main/kotlin/build/buf/protocgen/connect/internal/ProtoHelpers.kt

```diff
@@ -0,0 +1,476 @@
+// Copyright 2022-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buf.protocgen.connect.internal
+
+import com.google.protobuf.DescriptorProtos
+import com.google.protobuf.Descriptors
+
+/**
+ * Various helper methods, particularly useful when generating Java code.
+ * Original code from: https://codereview.appspot.com/912042.
+ *
+ * @author t.broyer@ltgt.net Thomas Broyer
+ * Based on the initial work of:
+ *  @author kenton@google.com Kenton Varda
+ * Based on original Protocol Buffers design by
+ *  Sanjay Ghemawat, Jeff Dean, and others.
+ */
+/**
+ * Name of the outer class scope insertion point.
+ */
+internal fun outerClassScopeInsertionPoint(): String = "outer_class_scope"
+
+/**
+ * Returns the insertion point name for a given message's class scope.
+ */
+internal fun getClassScopeInsertionPoint(descriptor: Descriptors.Descriptor): String {
+    return "class_scope:" + descriptor.fullName
+}
+
+/**
+ * Returns the insertion point name for a given message's builder class scope.
+ */
+internal fun getBuilderScopeInsertionPoint(descriptor: Descriptors.Descriptor): String {
+    return "builder_scope:" + descriptor.fullName
+}
+
+/**
+ * Returns the insertion point name for a given enum's scope.
+ */
+internal fun getEnumScopeInsertionPoint(descriptor: Descriptors.EnumDescriptor): String {
+    return "enum_scope:" + descriptor.fullName
+}
+
+/**
+ * Commonly-used separator comments (a line of '=').
+ */
+internal const val THICK_SEPARATOR = "// ===================================================================\n"
+
+/**
+ * Commonly-used separator comments (a line of '-').
+ */
+internal const val THIN_SEPARATOR = "// -------------------------------------------------------------------\n"
+
+/**
+ * Parses a set of comma-delimited name/value pairs.
+ *
+ *
+ * Several code generators treat the parameter argument as holding a list of
+ * options separated by commas: e.g., `"foo=bar,baz,qux=corge"` parses
+ * to the pairs: `("foo", "bar"), ("baz", ""), ("qux", "corge")`.
+ *
+ *
+ * When a key is present several times, only the last value is retained.
+ */
+internal fun parseGeneratorParameter(
+    text: String
+): Map<String, String> {
```

Comment:
I think you've written something that is redundant with this. So you could probably just use this version (or replace this with what you wrote).